### PR TITLE
redis: Add some type hints

### PR DESCRIPTION
**Pull request by bochecha**
Note that I only tested `brpop` and `lpush`, but I figured I'd do the other two as well since they are symmetric.

### CODE REVIEW

**Comment by bochecha** [comment]
As discussed on Gitter, I fixed the types for the "string-like" parameters.

File: third_party/2and3/redis/client.pyi [hunk 1]
```diff
@@ -161,22 +163,22 @@ class Redis(object):
     def type(self, name): ...
     def watch(self, *names): ...
     def unwatch(self): ...
-    def blpop(self, keys, timeout=...): ...
-    def brpop(self, keys, timeout=...): ...
+    def blpop(self, keys: Union[_Str, Sequence[_Str]], timeout: int = ...) -> Optional[Tuple[bytes, bytes]]: ...
+    def brpop(self, keys: Union[_Str, Sequence[_Str]], timeout: int = ...) -> Optional[Tuple[bytes, bytes]]: ...
```
Comment (srittau) [inline third_party/2and3/redis/client.pyi]: `Sequence` -> `Iterable`, since it's more general.

Comment (bochecha) [reply third_party/2and3/redis/client.pyi]: Indeed. Fixed in the latest push to this branch.

**Comment by srittau** [comment]
Will merge when tests pass. Just for future reference: Please don't force-push as it makes it harder to review changes.

**Comment by bochecha** [comment]
> Just for future reference: Please don't force-push as it makes it harder to review changes.

Oh, ok. Other projects I contribute to tend to prefer force pushing so the merged history is clean, but I'm happy to adapt. :slightly_smiling_face: 

Thanks for the review and for the help!

**Pull request merged**